cfme/infrastructure/provider/kubevirt.py

```diff
@@ -0,0 +1,45 @@
+from cfme.utils.providers import get_crud
+from cfme.containers.provider.openshift import VirtualizationEndpoint
+from . import InfraProvider
+
+
+class KubeVirtProvider(InfraProvider):
+
+    def __init__(self, name=None, endpoints=None, key=None, provider_data=None,
+                 parent_provider=None, appliance=None):
+
+        self.parent_provider = parent_provider
+
+        super(KubeVirtProvider, self).__init__(
+            name=name,
+            endpoints=endpoints,
+            key=key,
+            provider_data=provider_data,
+            appliance=appliance)
+
+    def create(self, *args, **kwargs):
+        # Override the standard behaviour to actually create the parent container provider first.
+        created = self.parent_provider.create()
+        return created
+
+    @classmethod
+    def from_config(cls, prov_config, prov_key, appliance=None):
+
+        endpoints = {}
+        token_creds = cls.process_credential_yaml_key(prov_config['credentials'], cred_type='token')
+        for endp in prov_config['endpoints']:
+            if VirtualizationEndpoint.name == endp:
+                prov_config['endpoints'][endp]['token'] = token_creds.token
+                endpoints[endp] = VirtualizationEndpoint(**prov_config['endpoints'][endp])
+
+        parent_provider = get_crud('ocp-kubevirt')
```

Comment:
absolutely cannot hardcode this provider name here.

Reply:
Your yaml for this provider will need to include a map of the parent provider key, which you'll call `get_crud` against here, and then pass as `parent_provider` to the constructor.

Reply:
suggestion for more robust way?

Reply:
yaml must have a parent_provider map, raise exception if it doesn't.  Use that map ( parent_provider: "ocp-kubevirt" ) to call `providers.get_crud(prov_config['parent_provider'])`

Reply:
corrected.